azure-devops/create-new-agent-image.ps1

```diff
@@ -190,18 +190,36 @@ New-AzVm `
 ####################################################################################################
 Write-Progress `
   -Activity $ProgressActivity `
-  -Status 'Running provisioning script in VM' `
+  -Status 'Running provisioning script provision-image.ps1 in VM' `
   -PercentComplete (100 / $TotalProgress * $CurrentProgress++)
 
-$VM = Get-AzVM -ResourceGroupName $ResourceGroupName -Name $ProtoVMName
-$PrototypeOSDiskName = $VM.StorageProfile.OsDisk.Name
 Invoke-AzVMRunCommand `
   -ResourceGroupName $ResourceGroupName `
   -VMName $ProtoVMName `
   -CommandId 'RunPowerShellScript' `
   -ScriptPath "$PSScriptRoot\provision-image.ps1" `
   -Parameter @{AdminUserPassword = $AdminPW }
 
+####################################################################################################
+Write-Progress `
+  -Activity $ProgressActivity `
+  -Status 'Restarting VM' `
+  -PercentComplete (100 / $TotalProgress * $CurrentProgress++)

```

Comment:
Would `100 * $CurrentProgress++ / $TotalProgress` be clearer? Ditto below. Actually, this appears to be repeated for each step - should it be extracted as a PowerShell function so only the status string needs to be passed?

Reply:
I don't know it's a pretty simple expression. I also find the way it is written to be more obvious than your suggestion.